schema/messageformat2/test_schema.json

```diff
@@ -0,0 +1,104 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "title": "Data-driven tests",
+  "type": "object",
+  "additionalProperties": false,
+  "required": ["scenario", "testType", "tests"],
+  "properties": {
+    "scenario": {
+      "type": "string",
+      "description": "Identifier for the tests in the file."
+    },
+    "description": {
+      "type": "string",
+      "description": "Information about the test scenario."
+    },
+    "testType": {
+      "type": "string",
+      "enum": ["syntax"],
+      "description": "A tag for the executor to decide how to run the test."
+    },
+    "tests": {
+      "type": "array",
+      "items": { "$ref": "#/$defs/test" },
+      "minItems": 1
+    }
+  },
+  "$defs": {
+    "test": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["label", "locale", "pattern"],
+      "properties": {
+        "label": {
+          "type": "string",
+          "description": "Identifier for the test."
+        },
+        "description": {
+          "type": "string",
+          "description": "Information about the test."
+        },
+        "locale": {
+          "type": "string"
+        },
+        "pattern": {
+          "type": "string"
+        },
+        "inputs": {
+          "type": "array",
+          "items": { "$ref": "#/$defs/arg" }
+        },
+        "verify": {
+          "type": "string"
+        }
+      }
+    },
+    "arg": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["type", "value"],
+      "properties": {
+        "name": {
+          "type": "string"
+        },
+        "type": {
+          "type": "string",
+          "enum": ["string", "number", "datetime"]
+        },
```

Comment:
I don't think `datetime` is a type in JSON schema.

But furthermore, it seems arbitrary to allow only string and number types to be passed as is, only because that's what JSON is capable of representing. string can behave as a lowest common denominator type when you aren't sure if the conduit or the receiving end of passing data can support the more structured type that the sending end is using. 

This isn't our problem to solve here or now, but it brings up the question whether formatting functions should also be required to implement a function that parse (deserialize from string) input values. If our tests want to pass a PersonName object to the `:person` formatter function, how do we represent that in the test input, and how do our executor implementations per library/language/platform parsed the serialized JSON that they receive?